node/network/availability-recovery/src/lib.rs

```diff
@@ -943,12 +1019,21 @@ async fn launch_recovery_task<Context>(
 
 	let phase = backing_group
 		.and_then(|g| session_info.validator_groups.get(g))
-		.map(|group| Source::RequestFromBackers(RequestFromBackers::new(group.clone())))
+		.map(|group| {
+			Source::RequestFromBackers(RequestFromBackers::new(
+				group.clone(),
+				erasure_task_tx.clone(),
+			))
+		})
 		.unwrap_or_else(|| {
-			Source::RequestChunks(RequestChunksFromValidators::new(params.validators.len() as _))
+			Source::RequestChunks(RequestChunksFromValidators::new(
+				params.validators.len() as _,
+				erasure_task_tx.clone(),
+			))
 		});
 
-	let recovery_task = RecoveryTask { sender: ctx.sender().clone(), params, source: phase };
+	let recovery_task =
+		RecoveryTask { sender: ctx.sender().clone(), params, source: phase, erasure_task_tx };
 
 	let (remote, remote_handle) = recovery_task.run().remote_handle();
 
```

Comment:
Wouldn't using spawn_blocking here, have the same effect ?
```
if let Err(e) = ctx.spawn("recovery-task", Box::pin(remote)) {
		gum::warn!(
			target: LOG_TARGET,
			err = ?e,
			"Failed to spawn a recovery task",
		);
	}
```

Reply:
Not really as it would be much longer lived, for the entire duration of the availability data fetching which is I/O only. We want to use a thread only for CPU intensive work.